### DISCUSSION TITLE
[TS/V7]How to reuse existing components

### DISCUSSION THREAD

**Discussion by matamatanot** [react-hook-form/react-hook-form#4739] [Q&A] [answered]
https://codesandbox.io/s/react-hook-form-useform-template-forked-rtskc?file=/src/index.tsx

How can I reuse FirstName's `Input` Component?
`Input` is used from Form1 and Form2.

**Comment by kotarella1110** [comment, marked as answer]
@matamatanot Sorry for the late reply 🙏🏻 
I think you can solve the problem by using Generics. 

https://codesandbox.io/s/4739-jbg56?file=/src/index.tsx
